grammar.pegjs

```diff
@@ -88,8 +88,12 @@ concatenation "concatenation of strings and/or variables"
           / commandSubstitution
           / singleQuote
           / doubleQuote
+          / equal
           )+
 
+equal "equal"
```

Comment:
Also, the `$` should be allowed in this context: `echo foo$` should succeed (`$` is a literal dollar sign, no special meaning). It also succeeds for the edge case `echo $`.

Also, in `sh`, `echo !foo` works (literal `!foo`). In `bash`, `!` has a special meaning, so this should say `!foo: event not found`. Both support `echo !`. It's probably fine to start with `sh` support and add in bash support later.